src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.cpp

```diff
@@ -0,0 +1,279 @@
+
+    ChipLogProgress(Crypto, "SE05x: RemoveOpKeypair for Fabric %02x", fabricIndex);
+
+    // Remove pending state if matching
+    if ((mPendingKeypair != nullptr) && (fabricIndex == mPendingFabricIndex))
+    {
+        RevertPendingKeypair();
+    }
+
+    while (i++ < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        if (keyidFabIdMapping[i].fabricIndex == fabricIndex)
+        {
+            // Delete the keyPair associated with the fabric
+            keyidFabIdMapping[i].isPending = 0;
+            Platform::Delete<Crypto::P256KeypairHSM>(keyidFabIdMapping[i].pkeyPair);
+            keyidFabIdMapping[i].pkeyPair    = NULL;
+            keyidFabIdMapping[i].keyId       = 0;
+            keyidFabIdMapping[i].fabricIndex = 0;
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void PersistentStorageOperationalKeystoreHSM::RevertPendingKeypair()
+{
+    ChipLogProgress(Crypto, "SE05x: RevertPendingKeypair");
+    VerifyOrReturn(mStorage != nullptr);
+    // Just reset the pending key, we never stored anything
+    ResetPendingKey();
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::SignWithOpKeypair(FabricIndex fabricIndex, const ByteSpan & message,
+                                                                      Crypto::P256ECDSASignature & outSignature) const
+{
+    uint8_t i = 0;
+    ChipLogProgress(Crypto, "SE05x: SignWithOpKeypair");
+
+    if (mIsPendingKeypairActive && (fabricIndex == mPendingFabricIndex))
+    {
+        VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INTERNAL);
+        // We have an override key: sign with it!
+        ChipLogProgress(Crypto, "SE05x: SignWithOpKeypair ==> using mPendingKeypair");
+        return mPendingKeypair->ECDSA_sign_msg(message.data(), message.size(), outSignature);
+    }
+    else
+    {
+        while (i++ < MAX_KEYID_SLOTS_FOR_FABRICS)
+        {
+            if ((keyidFabIdMapping[i].fabricIndex == fabricIndex) && (keyidFabIdMapping[i].isPending == 0))
+            {
+                ChipLogProgress(Crypto, "SE05x: SignWithOpKeypair ==> using stored keyPair");
+                return keyidFabIdMapping[i].pkeyPair->ECDSA_sign_msg(message.data(), message.size(), outSignature);
+            }
+        }
+    }
+
+    ChipLogProgress(Crypto, "SE05x: SignWithOpKeypair ==> No keyPair found");
+    return CHIP_ERROR_INTERNAL;
+}
+
+Crypto::P256Keypair * PersistentStorageOperationalKeystoreHSM::AllocateEphemeralKeypairForCASE()
+{
+    ChipLogProgress(Crypto, "SE05x: AllocateEphemeralKeypairForCASE using se05x");
+    Crypto::P256KeypairHSM * pkeyPair = Platform::New<Crypto::P256KeypairHSM>();
+    pkeyPair->SetKeyId(kKeyId_case_ephemeral_keyid);
+    return pkeyPair;
+}
+
+void PersistentStorageOperationalKeystoreHSM::ReleaseEphemeralKeypair(Crypto::P256Keypair * keypair)
+{
+    ChipLogProgress(Crypto, "SE05x: ReleaseEphemeralKeypair using se05x");
+    Platform::Delete<Crypto::P256Keypair>(keypair);
```

Comment:
Does this not need to delete it as `P256KeypairHSM`?  Please double-check.

Reply:
Deleting it as P256KeypairHSM now.